### ISSUE TITLE
Remix Docker Issues

### ISSUE THREAD

**Issue by lauhon** [oven-sh/bun#7625]
### What version of Bun is running?

1.0.14+d8be3e51b

### What platform is your computer?

Darwin 23.1.0 arm64 arm

### What steps can reproduce the bug?

1. bun create remix
2. Change `renderToPipeableStream` import in `entry.server.tsx` to 
	```import { renderToPipeableStream } from "react-dom/server.node";```
 (Without this the docker build is crashing. That also comes up in [this Issue](https://github.com/oven-sh/bun/issues/5664))
5. Add "normal" Dockerfile:
	

<details>
<summary>Dockerfile</summary>

```Dockerfile
FROM oven/bun:1 as base
WORKDIR /usr/src/app

# Install dependencies and cache
FROM base AS install
RUN mkdir -p /temp/dev
COPY package.json bun.lockb /temp/dev/
RUN cd /temp/dev && bun install --frozen-lockfile --ignore-scripts 


# install with --production (exclude devDependencies)
RUN mkdir -p /temp/prod
COPY package.json bun.lockb /temp/prod/
RUN cd /temp/prod && bun install --frozen-lockfile --production --ignore-scripts 

# Copy node modules from dev to prod
FROM base AS prerelease
COPY --from=install /temp/dev/node_modules node_modules
COPY . .

ENV NODE_ENV=production
RUN bun run build

FROM base AS release
ENV NODE_ENV=production

COPY --from=install /temp/prod/node_modules node_modules
COPY --from=prerelease /usr/src/app/build build
COPY --from=prerelease /usr/src/app/public public
COPY --from=prerelease /usr/src/app/package.json .
ADD . .


CMD [ "bun", "start" ]
```

</details>

6.  ```docker build -t my-app . ``` &   ```docker run -p 3000:3000 my-app``` 

 

### What is the expected behavior?

Website is working in Browser

Remix log

```log
$ remix-serve ./build/index.js
[remix-serve] http://localhost:3000/ (http://172.17.0.2:3000/)
Expected a Response object
GET / - - - 14.717 ms
```

### What do you see instead?

When visiting localhost the browser behaves very strangely, if I visit localhost on a tab that has another page open, the request just refreshes the old page. If I try to visit localhost from a new tab, it just clears the address bar and does nothing.

**Comment by lauhon** [comment]
Hey @Electroid, did you have time to investigate this?

Would really love to use bun for a small prod app at my job, but this is holding me back.

**Comment by zackify** [comment]
yeah this and prisma generate not working in docker are huge bummers for me... if these two issues were solved (https://github.com/oven-sh/bun/issues/5320) then i could actually use bun after spending a few hours making tweaks to fix other problems... its too bad because bun is so fast but a lot of common libraries just dont work

**Comment by QuentinDutot** [comment]
Having same problem. It seems related with : https://github.com/oven-sh/bun/issues/9949

**Comment by davidgamero** [comment]
bumping- same issue here
